XenAdmin/Controls/SrPicker.cs

```diff
@@ -199,6 +208,24 @@ public void SetAffinity(Host host)
 			refresh();
 		}
 
+        /// <summary>
+        /// Returns how much disk space is required to create the disk on an SR
+        /// This depends on whether the SR is thin provisioned and what the initial allocation rate is
+        /// </summary>
+        /// <param name="sr"></param>
+        /// <returns></returns>
+        private long GetRequiredDiskSizeForSR(SR sr)
+        {
+            if (sr != null && sr.sm_config != null
+                && sr.sm_config.ContainsKey("allocation") && sr.sm_config["allocation"] == "dynamic")

```

Comment:
The check that SR is thinly provisioned can be extracted in a  SR read-only property, as it is used in many places.